### ISSUE TITLE
Add DDP support for observer and fake-quant modules

### ISSUE THREAD

**Issue by raghuramank100** [pytorch/pytorch#33648]
Current QAT implementation does not support DDP, this is because observers contain attributes that are not tensors. The solution is to make attributes of observers buffers so that they are broadcast across devices, much like batch norm running mean and variance.

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

**Comment by raghuramank100** [comment]
@supriyar Can we close this issue now?
